cpp/src/arrow/compute/api_scalar.h

```diff
@@ -53,16 +52,10 @@ class ARROW_EXPORT ElementWiseAggregateOptions : public FunctionOptions {
   bool skip_nulls;
 };
 
-/// Functor to calculate hash of an enum.
-template <typename T, typename R = typename std::underlying_type<T>::type,
-          typename = std::enable_if<std::is_enum<T>::value>>
-struct EnumHash {
-  R operator()(T val) const { return static_cast<R>(val); }
-};
-
 /// Rounding and tie-breaking modes for round compute functions.
 /// Additional details and examples are provided in compute.rst.
 enum class RoundMode : int8_t {
+  // Note: The HALF values need to be last and the first HALF entry is HALF_DOWN.
```

Comment:
By having the HALF_XXX enums "grouped" to one end of the enum, it allows for checking if the rounding mode is tie-breaking by simply comparing against one instead of all the cases.
```
if (rmode >= RoundMode::HALF_DOWN) { ... }
// vs
if (rmode == RoundMode::HALF_DOWN || rmode == RoundMode::HALF_UP || ... 4 more times) { ... }
```
I added asserts in the Round/MRound kernel implementations, but not sure where else I can place them? Round/MRound constructor?

Reply:
Actually, I think the asserts for enum order would fit best in the Round/MRoundOptions constructors.